hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/TcpIpJoiner.java

```diff
@@ -139,6 +142,7 @@ private void joinViaPossibleMembers() {
 
             while (shouldRetry() && (Clock.currentTimeMillis() - startTime < maxJoinMillis)) {
                 tryJoinAddresses(possibleAddresses);
+                rethrowProtocolException(possibleAddresses);
```

Comment:
This doesn't seem correct. Imagine for instance the following case:

```
    public static void main(String[] args) {
        {
            Config config = new Config();
            AdvancedNetworkConfig an = config.getAdvancedNetworkConfig().setEnabled(true);
            an.getJoin().getTcpIpConfig().setEnabled(true).clear().addMember("127.0.0.1:5701");
            an.setMemberEndpointConfig(createServerSocketConfig(5701)).setClientEndpointConfig(createServerSocketConfig(6701));
            HazelcastInstance hz = Hazelcast.newHazelcastInstance(config);
        }
        {
            Config config = new Config();
            AdvancedNetworkConfig an = config.getAdvancedNetworkConfig().setEnabled(true);
            an.getJoin().getTcpIpConfig().setEnabled(true).clear().addMember("127.0.0.1:6701").addMember("127.0.0.1:5701");
            an.setMemberEndpointConfig(createServerSocketConfig(5702)).setClientEndpointConfig(createServerSocketConfig(6702));
            HazelcastInstance hz = Hazelcast.newHazelcastInstance(config);
        }
    }
```
Even if the first address in the member-list on the second member fail, the member should keep trying with the next one.

Reply:
I will change this rethrow logic from ANY to ALL

Reply:
Changed it. Now logs error message if few of them fail. It shutdowns like previously if all of the connections fail.

Reply:
Also added new tests related to this see 47a7ccd